app/services/flood_risk_engine/reference_number.rb

```diff
@@ -0,0 +1,58 @@
+module FloodRiskEngine
```

Comment:
I know what you mean, but my starting point was how it would be used:

```ruby
number = ReferenceNumber.with(
  prefix: "EXFRA",
  seed: id,
  offset: 10_000,
  padding: "0",
  minimum_length: 11
)
```

And as I thought there was a reasonable chanced that any of these options could be changed or dropped, I needed to make it flexible enough to handle that.

Also although it is only formatting a string, there is a lot of complication in there - adding the prefix, padding, and offset.